codec/codec.go

```diff
@@ -7,49 +7,76 @@ import (
 )
 
 type (
-	// Marshaler defines the interface module codecs must implement in order to support
-	// backwards compatibility with Amino while allowing custom Protobuf-based
-	// serialization. Note, Amino can still be used without any dependency on
-	// Protobuf. There are two typical implementations that fulfill this contract:
+	// Codec defines a functionality for serializing other objects.
+	// Users can defin a custom Protobuf-based serialization.
+	// Note, Amino can still be used without any dependency on Protobuf.
+	// SDK provides to Codec implementations:
 	//
 	// 1. AminoCodec: Provides full Amino serialization compatibility.
 	// 2. ProtoCodec: Provides full Protobuf serialization compatibility.
-	Marshaler interface {
-		BinaryMarshaler
-		JSONMarshaler
+	Codec interface {
+		BinaryCodec
+		JSONCodec
 	}
 
-	BinaryMarshaler interface {
-		MarshalBinaryBare(o ProtoMarshaler) ([]byte, error)
-		MustMarshalBinaryBare(o ProtoMarshaler) []byte
+	BinaryCodec interface {
+		// Marshal returns binary encoding of v.
```

Comment:
What is `v` in the comment?

Reply:
yeah should be `o`. @xavierlepretre would you like to create a quick PR?